src/sql/workbench/services/query/common/queryManagement.ts

```diff
@@ -278,19 +278,21 @@ export class QueryManagementService implements IQueryManagementService {
 			const result = [];
 			let start = rowData.rowsStartIndex;
 			this._logService.trace(`Getting ${rowData.rowsCount} rows starting from index: ${rowData.rowsStartIndex}.`);
+			let pageIdx = 1;
 			do {
 				const rowCount = Math.min(pageSize, rowData.rowsStartIndex + rowData.rowsCount - start);
-				this._logService.trace(`Paged Fetch - Getting ${rowCount} rows starting from index: ${start}.`);
+				this._logService.trace(`Page ${pageIdx} - Getting ${rowCount} rows starting from index: ${start}.`);
 				const rowSet = await runner.getQueryRows({
 					ownerUri: rowData.ownerUri,
 					batchIndex: rowData.batchIndex,
 					resultSetIndex: rowData.resultSetIndex,
 					rowsCount: rowCount,
 					rowsStartIndex: start
 				});
-				this._logService.trace(`Paged Fetch - Received ${rowSet.resultSubset.rows} rows starting from index: ${start}.`);
+				this._logService.trace(`Page ${pageIdx} - Received ${rowSet.resultSubset.rows.length} rows starting from index: ${start}.`);
```

Comment:
This is where the error was, I was logging the rowSet.resultSubset.rows instead of its length.